types/report-renderer.d.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+
+declare global {
+  module LH.Renderer {
+    class ReportRenderer {
+      constructor(lhr: LH.Result, options?: ReportRendererOptions);
+
+      getContainerEl(): HTMLElement;
+      getTopBarEl(): HTMLElement;
```

Comment:
These functions combined with

> if it decides to move the scoregauge and footer to different parts of the DOM, sure. 

concerns me a bit about how these APIs will be used. I was envisioning something more like 

```js
/** This function renders a live report into a `div` currently disconnected from the document and returns it. Manually mutating the contents of this element will have unexpected consequences as the UI state is managed and possibly re-rendered by Lighthouse. */
export function renderReportIntoElement(containerEl: HTMLElement, lhr: LH.Result, options: RenderOptions): void
/** This function renders report components into their respective containers. Manually mutating the contents of each container will have unexpected consequences as the UI state is managed and possibly re-rendered by Lighthouse. */
export function renderReportComponentsIntoElements(containers: ComponentLocations, lhr: LH.Result, options: RenderOptions): void

interface RenderOptions extends PaulsExistingOptions {}
interface ComponentLocations {
  topbarEl: HTMLElement
  containerEl: HTMLElement
  scoreGaugeContainerEl?: HTMLElement // sounds like we might need this?
  footerContainerEl?: HTMLElement // sounds like we might need this?
}
```

Is the "But that's on them." meant to say a client that mutates the returned element need to handle any potential future UI updates on their own?

Reply:
> Is the "But that's on them." meant to say a client that mutates the returned element need to handle any potential future UI updates on their own?

yup. i like what you said with `Manually mutating the contents of this element will have unexpected consequences as the UI state is managed and possibly re-rendered by Lighthouse` and think that applies to my proposal as well.

I don't think we need to provide this much control. It hasn't been mentioned, but we want to add the Runtime settings block to PSI. So that makes me want to not expose it as a separate thing. But then I just checked [treo and they expose the footer separately](https://treo.sh/demo/1/pages/6/runs/30620?interval=1month) in the DOM. So... I can get with that. 

Okay so given that, I'd do a variant of what you have.. but mostly exposing dom nodes instead of appending into provided ones.


```ts
renderFullReport(): HTMLElement;
renderReportComponents(): {topbarEl: HTMLElement, mainEl: HTMLElement, headerEl: HTMLElement, categoriesEl: HTMLElement, footerEl: HTMLElement}
```

These components needs to follow the convention we decide from the DOM inconsistency I explored in https://github.com/GoogleChrome/lighthouse/issues/12254#issuecomment-877520562.

Reply:
+1 to functions that return an element (as opposed to taking a containerEl and rendering into that)

Reply:
> I'd do a variant of what you have.. but mostly exposing dom nodes instead of appending into provided ones.

SGTM 👍